Sources/SwiftProtobuf/JSONDecoder.swift

```diff
@@ -74,6 +74,8 @@ internal struct JSONDecoder: Decoder {
     if scanner.skipOptionalArrayEnd() {
       return
     }
+    let count = try scanner.countArrayElementsQuickly()
+    value.reserveCapacity(count)
```

Comment:
Shouldn't all these use count + value.count also?

Reply:
Not for JSON.  Text format needs it to handle cases like this:
```
someKey: [1, 2, 3]
someKey: [4, 5, 6]
```
but JSON doesn't permit duplicate keys.

Reply:
Ah, right.